yarn-project/aztec-nr/aztec/src/types/address.nr

```diff
@@ -0,0 +1,49 @@
+struct AztecAddress {
+  address: Field
+}
+
+impl AztecAddress {
+  fn new(address: Field) -> Self {
+    Self {
+      address
+    }
+  }
+
+  fn serialize(self: Self) -> [Field; 1] {
+    [self.address]
+  }
+
+  fn deserialize(fields: [Field; 1]) -> Self {
+    Self {
+      address: fields[0]
+    }
+  }
+}
+
+struct EthereumAddress {
+  address: Field
+}
+
+impl EthereumAddress {
+  fn new(address: Field) -> Self {
+    // Check that it actually will fit. Spending a lot of constraints here :grimacing:
+    let bytes = address.to_be_bytes(32);
+    for i in 0..12 {
+        assert(bytes[i] == 0, "Value too large for SafeU120");
```

Comment:
The error message should probably not be SafeU120 here. Also a test can be added in here with just noir tests. like.


```rust
#[test]
fn test_address() {
  // Test that address is good.
}

#[test(should_fail)]
fn test_bad_address() {
  // Ensure that address is failing.
}
```

Reply:
tell me I copied blindly without telling me I copied blindly smh